crates/astria-sequencer/src/grpc/optimistic.rs

```diff
@@ -0,0 +1,151 @@
+use std::{
+    pin::Pin,
+    sync::Arc,
+};
+
+use astria_core::{
+    generated::sequencerblock::optimisticblock::v1alpha1::{
+        optimistic_block_service_server::OptimisticBlockService,
+        GetBlockCommitmentStreamRequest,
+        GetBlockCommitmentStreamResponse,
+        GetOptimisticBlockStreamRequest,
+        GetOptimisticBlockStreamResponse,
+    },
+    primitive::v1::RollupId,
+    Protobuf,
+};
+use tonic::{
+    codegen::tokio_stream::{
+        wrappers::ReceiverStream,
+        Stream,
+    },
+    Request,
+    Response,
+    Status,
+};
+use tracing::error;
+
+use crate::app::OptimisticBlockChannels;
+
+pub(crate) struct OptimisticBlockServer {
```

Comment:
I think this should not be surfaced all the way to the app. Leave this private to the `gRPC` module.

Also:
```suggestion
struct OptimisticBlockServiceImpl {
```